src/app/verification-addresses/services/iva.service.ts

```diff
@@ -128,11 +186,27 @@ export class IvaService {
     }
     return this.#http
       .post<{ id: string }>(this.#userIvasUrl(userId), { type, value })
-      .pipe(map(({ id }) => id));
+      .pipe(
+        map(({ id }) => id),
+        tap((id) => this.#addIvaLocally({ id, type, value, userId })),
+      );
+  }
+
+  /**
+   * Delete an IVA locally
+   * @param ivaId - the ID of the IVA to delete
+   */
+  #deleteIvaLocally(ivaId: string): void {
```

Comment:
Can't we have a function to do local addition/state modification of IVAs as well? Or just use the update also for deletion?

Reply:
We have that already, in #addIvaLocally and #updateIvaStateLocally. The latter is shared by the different state modification methods.

Reply:
Ah no I meant having a single update method instead of all these separate ones that do similar things.

Reply:
Ok. But I think they are distinct enough to warrant their own methods. Also, I think it's good practice to have smaller functions that do only one thing (aka Curly's Law or Single Responsibility Principle):
- https://activeprogrammerhq.com/functions-clean-code/